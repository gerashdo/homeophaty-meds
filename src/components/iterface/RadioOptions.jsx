import PropTypes from "prop-types"

import './radio-options.css'

export const RadioOptions = ({ options, value, onChange = () => {} }) => {

    return (
        <div 
            className="radio-options"
        >
            {
                options.map( option => (
                    <label
                        key={ option }
                        className="radio-input-container"
                    > 
                        <input 
                            type="radio"
                            id={ option }
                            value={ option }
                            checked={ value === option }
                            onChange={ onChange }
                        />
                        <label htmlFor={ option }>{ option }</label>
                    </label>
                ))
            }
        </div>
    )
}

RadioOptions.propTypes = {
    options: PropTypes.arrayOf( PropTypes.string ).isRequired,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func,
}
