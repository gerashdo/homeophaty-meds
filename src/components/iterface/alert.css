
.alert-container{
    align-items: center;
    background-color: var(--color-primary-darker);
    border-radius: .75rem;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
    color: white;
    display: flex;
    gap: 1rem;
    position: fixed;
    top: 50px;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 1.5rem 3rem 1.5rem 1.5rem;
    overflow: hidden;
}

.alert-container span {
    font-size: 1.5rem;
}

.alert-container.success {
    background-color: var(--color-secondary);
}

.alert-container.error {
    background-color: var(--color-negative);
}

.alert-container .close {
    position: absolute;
    top: 5px;
    right: 10px;
    border-radius: .25rem;
    cursor: pointer;
    font-size: var(--font-normal);
    padding: 0.125rem 0.5rem;
    transition: all 0.2s ease;
    opacity: 0.7;
}

.alert-container .close:hover {
    opacity: 1;
}