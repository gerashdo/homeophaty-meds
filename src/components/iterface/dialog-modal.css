
dialog{
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

dialog.dialog-modal{
    border: 0;
    border-radius: .75rem;
    /* display: flex; */
    /* flex-direction: column; */
    /* gap: 1rem; */
    padding: 1.5rem 2rem;
    box-shadow: 0 5px 10px rgba(0, 0, 0, 0.1);
}

dialog.dialog-modal::backdrop{
    background: rgba(0, 0, 0, 0.3);
}

dialog.dialog-modal .title h4{
    font-weight: var(--font-bold);
    font-size: var(--font-size-subtitle);
    text-align: start;
    margin-bottom: 1rem;
}

dialog.dialog-modal .content {
    margin-bottom: 2rem;
}

dialog.dialog-modal .footer{
    display: flex;
    justify-content: flex-end;
    gap: 1rem;
}