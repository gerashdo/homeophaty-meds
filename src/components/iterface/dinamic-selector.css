
.dinamic-selector{
    background-color: var(--color-gray-background);
    border-radius: 0 0px 10px 10px;
    box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.2);
    position: absolute;
    width: 100%;
    max-height: 200px;
    overflow: auto;
    top: 40px;
}

.dinamic-selector ul {
    list-style: none;
}

.dinamic-selector ul li {
    padding: 0.5rem 1rem;
    cursor: pointer;
}

.dinamic-selector ul li:hover {
    background-color: var(--color-primary-lighter);
    color: white;
}