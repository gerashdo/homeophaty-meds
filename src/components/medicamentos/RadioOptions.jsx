
import './radio-options.css'

export const RadioOptions = ({ options, value, onChange }) => {

    return (
        <div 
            className="radio-options"
        >
            {
                options.map( option => (
                    <label
                        key={ option }
                        className="radio-input-container"
                    > 
                        <input 
                            type="radio"
                            id={ option }
                            value={ option }
                            checked={ value === option }
                            onChange={ onChange }
                        />
                        <label htmlFor={ option }>{ option }</label>
                    </label>
                ))
            }
        </div>
    )
    }
