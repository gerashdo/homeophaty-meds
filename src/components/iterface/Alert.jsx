import { useDispatch, useSelector } from "react-redux"
import { BsFillCheckCircleFill } from "react-icons/bs"

import './alert.css'
import { useEffect } from "react"
import { stopAlert } from "../../store/slices/ui"

export const Alert = ({ children }) => {
    const dispatch = useDispatch()
    const { alertActive, alertMessage, alertType } = useSelector( state => state.ui )

    const closeAlert = () => {
        dispatch( stopAlert() )
    }

    useEffect(() => {
        if( alertActive ){
            setTimeout( () => {
               closeAlert()
            }, 8000)
        }
    }, [ alertActive ])

    return (
        <>
        { children }

        {
            alertActive
            ?(<div className={`alert-container ${alertType}`}>
                <span>
                    <BsFillCheckCircleFill />
                </span>
                <p>{ alertMessage }</p>
                <span 
                    className="close"
                    onClick={ closeAlert }
                >x</span>
            </div>)
            : null
        }
        </>
    )
}
