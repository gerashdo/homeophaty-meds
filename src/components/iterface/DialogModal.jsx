
import { useEffect, useRef } from 'react'
import './dialog-modal.css'

export const DialogModal = ({ title, content, open, onClose, onAccept }) => {
  const ref = useRef(null)

  useEffect(() => {
    if (open) {
      ref.current?.showModal()
    } else {
      ref.current?.close()
    }
  }, [open])

  const handleAccept = () => {
    onAccept()
    onClose()
  }

  return (
    <dialog
      ref={ref}
      className='dialog-modal'
    >
      <header className='title'>
        <h4>{title}</h4>
      </header>

      <p
        className='content'
      >
        {content}
      </p>
      <footer
        className='footer'
      >
        <button
          onClick={() => onClose()}
          className='simple'
        >
          Cancelar
        </button>
        <button
          onClick={handleAccept}
          className='primary'
        >
          Aceptar
        </button>
      </footer>
    </dialog>
  )
}
