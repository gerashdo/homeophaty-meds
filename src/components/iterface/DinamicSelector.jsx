
import { useEffect, useState } from 'react'
import { searchStringInMed } from '../../helpers'
import PropTypes from 'prop-types'

import './dinamic-selector.css'

export const DinamicSelector = ({ resetValues, valueForFilter, onSetValue }) => {
  const [valuesForUse, setValuesForUse] = useState(resetValues)

  useEffect(() => {
    if (valueForFilter) {
      setValuesForUse(searchStringInMed(valueForFilter, resetValues))
    } else {
      setValuesForUse(resetValues)
    }
  }, [valueForFilter, resetValues])

  const setSelectedValue = ({ target }) => {
    onSetValue(target.innerText)
  }

  return (
    <div
      className='absolute bg-white rounded-md shadow z-20 w-72 inset-x-0 max-h-60 top-16'
    >
      <ul>
        {
          valuesForUse.length > 0
            ? valuesForUse.map(med => (
              <li
                key={med._id}
                onMouseDown={setSelectedValue}
                className='hover:bg-gray-200 cursor-pointer p-2'
              >
                {med.name} {med.ch}
              </li>
            ))
            : <li data-testid='no-results'>Sin resultados</li>
        }
      </ul>
    </div>
  )
}

DinamicSelector.propTypes = {
  resetValues: PropTypes.array.isRequired,
  valueForFilter: PropTypes.string,
  onSetValue: PropTypes.func.isRequired
}
