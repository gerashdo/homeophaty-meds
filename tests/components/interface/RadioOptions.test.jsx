import { fireEvent, render, screen } from "@testing-library/react";
import { RadioOptions } from "../../../src/components/iterface/RadioOptions";

describe('Tests for <RadioOptions />', () => {
    const options = [ 'a', 'b', 'c', 'd' ]
    const onChange = jest.fn()
    

    test('should render the component correctly', () => {
        const value = ''
        const { container } = render(
            <RadioOptions 
                options={ options }
                value={ value }
            />
        )

        expect( screen.queryAllByRole('radio').length ).toBe( options.length )
        expect( container ).toMatchSnapshot()
    });

    test('should set as checked the option that is equal to the value', () => {
        let value = options[ 0 ]
        const { rerender } = render(
            <RadioOptions 
                options={ options }
                value={ value }
            />
        )

        const optionOne = screen.getByRole('radio', { name: options[ 0 ] })
        const optionTwo = screen.getByRole('radio', { name: options[ 1 ] })
        expect( optionOne ).toBeChecked()
        expect( optionTwo ).not.toBeChecked()

        
        value = options[ 1 ]
        rerender(
            <RadioOptions 
                options={ options }
                value={ value }
            />
        )

        expect( optionOne ).not.toBeChecked()
        expect( optionTwo ).toBeChecked()
    });

    test('should call onChange function when a option is clicked', () => {
        const value = options[ 0 ]
        render(
            <RadioOptions 
                options={ options }
                value={ value }
                onChange={ onChange }
            />
        )
        
        fireEvent.click( screen.getByRole( 'radio', { name: options[ 1 ] } ) )
        expect( onChange ).toHaveBeenCalled()
    });

});