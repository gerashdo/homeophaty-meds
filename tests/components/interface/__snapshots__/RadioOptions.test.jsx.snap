// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <RadioOptions /> should render the component correctly 1`] = `
<div>
  <div
    class="radio-options"
  >
    <label
      class="radio-input-container"
    >
      <input
        id="a"
        type="radio"
        value="a"
      />
      <label
        for="a"
      >
        a
      </label>
    </label>
    <label
      class="radio-input-container"
    >
      <input
        id="b"
        type="radio"
        value="b"
      />
      <label
        for="b"
      >
        b
      </label>
    </label>
    <label
      class="radio-input-container"
    >
      <input
        id="c"
        type="radio"
        value="c"
      />
      <label
        for="c"
      >
        c
      </label>
    </label>
    <label
      class="radio-input-container"
    >
      <input
        id="d"
        type="radio"
        value="d"
      />
      <label
        for="d"
      >
        d
      </label>
    </label>
  </div>
</div>
`;
